### PR TITLE
Add support for custom schema prototype

### PR DESCRIPTION
**Pull request by cschiewek**
Currently, the only way to add custom schema directives in Absinthe is to define a schema prototype.  However, absinthe only supports a single prototype, so using schema directives from this library, and your own custom schema prototype, isn't possible.

This PR attempts to solve that by:
- Adding `:skip_prototype` option to `Absinthe.Federation.Schema`
- Bypasses the `@prototype_schema` module attribute when said option is true
- Moves all the directives (and associated types) to a new module, so that they can be included in both the Absinthe Federations default prototype or any dependants prototype.

I also added a test and updated the README with instructions on how to configure when downstream projects already have a prototype.

Please let me know if there's an alternative approach you'd prefer.

### CODE REVIEW

**Comment by kzlsakal** [comment]
@cschiewek nice! I was thinking about using Absinthe's `import_directives` macro (which was made available with 1.7.1) in the future. This is a great workaround for the current implementation.

File: README.md [hunk 1]
```diff
@@ -42,6 +42,31 @@ defmodule MyApp.MySchema do
 end
 ```
 
+If you are already using a schema prototype
```
Comment (kzlsakal) [inline README.md:45]: What do you think about moving this under the "Usage" right below "Macro based schemas" example?

File: lib/absinthe/federation/schema/prototype/directives.ex [hunk 2]
```diff
@@ -0,0 +1,146 @@
+defmodule Absinthe.Federation.Schema.Prototype.Directives do
```
Comment (kzlsakal) [inline lib/absinthe/federation/schema/prototype/directives.ex:1]: I think we could rename this to `Absinthe.Federation.Schema.FederatedDirectives`, what do you think?

**Review by kzlsakal — CHANGES_REQUESTED** [review]

**Comment by cschiewek** [comment]
> @cschiewek nice! I was thinking about using Absinthe's `import_directives` macro (which was made available with 1.7.1) in the future. This is a great workaround for the current implementation.

I tried that first, `import_directives` only works with operation directives, not schema directives, so you can't use them.

File: lib/absinthe/federation/schema/prototype/directives.ex (hunk 2 above)
Comment (cschiewek) [reply lib/absinthe/federation/schema/prototype/directives.ex:1]: Agreed.  I'll update shortly.

File: README.md (hunk 1 above)
Comment (cschiewek) [reply README.md:45]: Will do!

**Comment by kzlsakal** [comment]
> > @cschiewek nice! I was thinking about using Absinthe's `import_directives` macro (which was made available with 1.7.1) in the future. This is a great workaround for the current implementation.
> 
> I tried that first, `import_directives` only works with operation directives, not schema directives, so you can't use them in a prototype module.

Good to know, and that's a bummer :D

**Comment by cschiewek** [comment]
@kzlsakal I've addressed all the feedback (TY!) and also added descriptions for the `_service` and `sdl` fields, as apollo's linter complained that they were missing.

File: lib/absinthe/federation/types.ex [hunk 3]
```diff
@@ -9,7 +9,16 @@ defmodule Absinthe.Federation.Types do
...
+  @desc """
+  Schema composition at the gateway requires having each service's schema, annotated with its federation configuration.
+  This information is fetched from each service using _service, an enhanced introspection entry point added to the
+  query root of each federated service.
+  """
   object :service, name: "_Service" do
+    @desc """
+    This SDL (schema definition language) is a printed version of the service's schema including the annotations of
+    federation directives. This SDL does not include the additions of the federation spec.
+    """
```
Comment (cschiewek) [inline lib/absinthe/federation/types.ex:21]: There descriptions were pulled from https://www.apollographql.com/docs/federation/v1/federation-spec/#federation-schema-specification

**Review by kzlsakal — APPROVED** [review]
Awesome contribution @cschiewek!

**Pull request merged**